gelbooru-overhaul.user.js

```diff
@@ -9,485 +9,1142 @@
+     */
+    async function applyTweakFastDLPost(value) {
+        if (currentPageType != utils.pageTypes.POST) return;
+
+        utils.debugLog(`Applying FastDLPost state: ${String(value)}`);
+
+        let post = document.querySelector("#gelcomVideoPlayer, #image");
+        if (value) {
+            post.addEventListener("contextmenu", downloadPostWithRMB);
+        } else {
+            post.removeEventListener("contextmenu", downloadPostWithRMB);
+        }
+    }
+    //      Infinite Scroll
+    /**
+     * @type {PreferenceUpdateCallback}
+     * @param {boolean} value
+     */
+    async function applyTweakInfiniteScroll(value) {
+        if (currentPageType != utils.pageTypes.GALLERY) return;
+
+        utils.debugLog(`Applying InfiniteScroll state: ${String(value)}`);
+
+        infiniteScrolling.setup(value);
+    }
+    /**
+     * @type {PreferenceUpdateCallback}
+     * @param {boolean} value
+     */
+    async function applyTweakPaginatorOnTop(value) {
+        if (currentPageType != utils.pageTypes.GALLERY) return;
+
+        utils.debugLog(`Applying InfiniteScroll state: ${String(value)}`);
+
+        if (value) {
+            if (document.querySelector(".top-pagination")) return;
+
+            /** @type {HTMLElement} */
+            let topPagination = document.querySelector(".pagination").cloneNode(true);
+            topPagination.classList.add("top-pagination");
+            document.querySelector("main").insertBefore(topPagination, document.querySelector(".thumbnail-container"));
+        } else {
+            document.querySelector(".top-pagination").remove();
+        }
+    }
+    /**
+     * @type {PreferenceUpdateCallback}
+     * @param {boolean} value
+     */
+    async function applyTweakGoToTop(value) {
+        if (currentPageType != utils.pageTypes.GALLERY) return;
+
+        utils.debugLog(`Applying InfiniteScroll state: ${String(value)}`);
+
+        if (value) {
+            let goTopDiv = document.createElement("div");
+            let goTopA = document.createElement("a");
+
+            goTopDiv.className = "alert alert-info";
+            goTopDiv.id = "go-top";
+            goTopDiv.addEventListener("click", () => window.scrollTo({ top: 0, behavior: 'smooth' }));
+
+            goTopA.textContent = "Go Top";
+
+            goTopDiv.appendChild(goTopA);
+            document.body.appendChild(goTopDiv);
+        } else {
+            document.querySelector("#go-top").remove();
+        }
+    }
+    //      Advanced Blacklist
+    /**
+     * @type {PreferenceUpdateCallback}
+     * @param {boolean} value
+     */
+    async function applyTweakAdvancedBlacklist(value) {
+        if (![utils.pageTypes.GALLERY, utils.pageTypes.POST].includes(currentPageType)) return;
+        utils.debugLog(`Applying AdvancedBlacklist state: ${String(value)}`);
+
+        blacklistManager.setupManager(value);
+    }
+    function applyVariableBlacklist(value) {
+        blacklistManager.orderEntriesByHitCount = value;
```

Comment:
```suggestion
    function applyVariableBlacklist(value) {
        if (![utils.pageTypes.GALLERY, utils.pageTypes.POST].includes(currentPageType)) return;
        blacklistManager.orderEntriesByHitCount = value;
```
Fixes a crash before the first contentful paint on pages without `#go-advBlacklistEntries`.